### ISSUE TITLE
Haskell packages should have cblrpm update run on them

### ISSUE THREAD

**Issue by mathstuf** [phracek/rebase-helper#110]
`cblrpm update` should be used to update Haskell packages before doing a scratch build. This will give the build a better chance to succeed since extra BuildRequires would automatically be added.

Cc: @juhp

**Comment by phracek** [comment]
Thanks for the bug and improvement.

Is this mentioned in %prep section?
How does it look like SPEC file?

rebase-helper has a functionality which provides building package via mock or even fedpkg.

**Comment by mathstuf** [comment]
`cblrpm` is a tool like `cpan2spec`. It generates spec files from Haskell's cabal package files including BuildRequirements (adding and removing). Though I think a diff which ignores the `%changelog` and any `%summary` or `%description` changes would be best (since these are sometimes reworded or tweaked and the diff from the wordwrap can get noisy).

**Comment by phracek** [comment]
Sorry but I did not catch how it is related with rebase-helper?
Rebase-helper is a tool which helps you with rebasing to new upstream release.

**Comment by mathstuf** [comment]
`cblrpm diff` gives a diff of what it would take to update a Haskell package to the latest version as well (given the cabal file or the source tarball). My normal workflow:

  - update version according to the bug;
  - download source tarball;
  - add changelog;
  - commit what I have;
  - run `cblrpm diff | colordiff` for sanity (it will find the tarball just fine);
  - run `cblrpm diff | git apply` to get the change;
  - use `git checkout -p` to pull out the bits that get overwritten (like `%changelog`);
  - `git commit --amend && git push && fedpkg build`.

**Comment by juhp** [comment]
Yeah, "cabal-rpm update" can be used to rebase a Haskell spec file to the latest version from Hackage.

**Comment by phracek** [comment]
If you don't mind could you please send a PR?

**Comment by thozza** [comment]
We discussed this and we will add possibility to implement hooks before specific actions (download of files, build, ...) Related to #186
